tests/unit/test_repos_repo.py

```diff
@@ -874,6 +874,39 @@ def test_tree_required_sha(self):
 
         assert self.session.get.called is False
 
+    def test_update_label(self):
+        """Verify the request for updating a label."""
+        with mock.patch.object(Repository, 'label') as mocked_label:
```

Comment:
I know the tests were using this previously, but I have been working very hard to eliminate instances where we mock out portions of the object currently being tested (in this case Repository objects). Could these be re-written to only care about mocking the session object (like the rest of the unit tests)?